### ISSUE TITLE
ConvPoint implementation with your Library

### ISSUE THREAD

**Issue by shangguan91** [rusty1s/pytorch_geometric#1401]
Dear Matt, 

Could you give me some hint about constructing the ConvPoint Architecture with your library?

**Comment by rusty1s** [comment]
I am not sure what you mean with `ConvPoint`. Which paper are you referring to?

**Comment by shangguan91** [comment]
https://github.com/aboulch/ConvPoint

Matthias Fey <notifications@github.com> 于 2020年7月5日周日 下午5:31写道：

**Comment by rusty1s** [comment]
This would look like the following, although I am sure there are same ways to get around the for-loop making use of batched matrix multiplication:
```python
class ConvPoint(MessagePassing):
    def __init__(self, ...):
        super(ConvPoint, self).__init__(aggr='mean')
        self.mlp = Sequential(Linear(3, ...), ..., Linear(..., kernel_size))
        self.weight = Parameter(torch.Tensor(kernel_size, in_channels, out_channels)

    def forward(self, x, edge_index, pos, centroid):
        return self.propagate(edge_index, x=x, pos=pos, centroid=centroid)

    def message(self, x_j, pos_j, centroid_i):
        norm = self.mlp(pos_j - centroid_i)
        out = 0
        for k in self.weight.size(0):
            out += x_j @ self.weight[k] * norm[:, k]
        return out
```

**Comment by shangguan91** [comment]
Firstly, thank you for the code suggestion. Since i am not coder, i would be more intereted in the implementation.  i have successfully implemented edgeconv, splineconv in your package. Now, i intend to implemented this ConvPoint,Light-ConvPoint and KPConv as well.  Would it be possbile that these networks being shown as one example in your library soon?

**Comment by rusty1s** [comment]
Sure, I think adding it to the library should not be too hard. If you are willing, you can implement a first draft and send it as a PR. That would help me a lot.

**Comment by tchaton** [comment]
@shangguan91 Have a look at : https://github.com/nicolas-chaulet/torch-points3d

**Comment by shangguan91** [comment]
i tried to implant his model, however, i have some problem of the forward pass 

> -forward() missing 1 required positional argument: 'input_pts'

[https://github.com/aboulch/ConvPoint/blob/c8c3c7dad53904d6724b87ca5c8651328ccfd371/networks/network_classif.py](url)
. Is there anyway to rewrite his code into your def forward(self, data) format, 

`def forward(self, data):
        pos, batch = data.pos, data.batch
        x1 = self.conv1(pos, batch)
        x2 = self.conv2(x1, batch)
        out = self.lin1(torch.cat([x1, x2], dim=1))
        out = global_max_pool(out, batch)
        out = self.mlp(out)
        return F.log_softmax(out, dim=1)`

**Comment by shangguan91** [comment]
> @shangguan91 Have a look at : https://github.com/nicolas-chaulet/torch-points3d

Thank you
